pkgs/applications/blockchains/monero-cli/default.nix

```diff
@@ -38,12 +38,11 @@ let
   trezor-common = fetchFromGitHub {
     owner = "trezor";
     repo = "trezor-common";
-    rev = "bc28c316d05bf1e9ebfe3d7df1ab25831d98d168";
-    hash = "sha256-F1Hf1WwHqXMd/5OWrdkpomszACTozDuC7DQXW3p6248=";
+    rev = "3a631b730bb6f91c27d41360b9b4ec063352593b";
```

Comment:
bc28c316d05bf1e9ebfe3d7df1ab25831d98d168 is the commit upstream is using. What is the reason for overriding it?

Reply:
Thanks for the comment.

I don't know what submodule commits the `v0.18.3.4` release used. How can you tell?

Since I did not find a way to lock submodules to a specific revision, I assumed that whatever commit of the submodule is latest at the time of *monero-cli* release is the one being used.

That said, I am not sure actually how I ended up on `3a63`, since that one was probably not the latest at the time of the v0.18.3.4 release. Still, it is over a year newer than the original commit. It's dated May 24th 2024, while the original commit is over a year older than that, March 6th 2023.

*monero-cli* was failing to build on my Mac M1 with the original commit, but it builds with this one.

If the v0.18.3.4 release used the original commit of this submodule, can you please tell me how I can identify that?

Reply:
> I don't know what submodule commits the v0.18.3.4 release used. How can you tell?

The tagged versions are pinned at this [commit](https://github.com/trezor/trezor-common/tree/bff7fdfe436c727982cc553bdfb29a9021b423b0), which is much older.

> monero-cli was failing to build on my Mac M1 with the original commit, but it builds with this one.

Ok, then it makes sense to update it.

> can you please tell me how I can identify that?

From github you can switch the current view to a tag using the the button just below the repository name. Then navigate into the `external/` directory and you will see the commits of these submodules.

Reply:
Thanks. By your instructions, I see it's at commit `bff7`, as you said in "are pinned at this commit, which is much older." It's four years old! But then I still don't understand how we arrived at commit `bc28`, which you wrote "is the commit upstream is using". Can you clarify where `bc28` came from?

Reply:
bc28 is used in master. I should have paid more attention and used the one from the tagged version, but it doesn't seem to be a problem.

Reply:
Thanks for explaining - mystery solved! I updated trezor-common to the tagged version in my latest rev.